src/averageAges.js

```diff
@@ -37,7 +43,21 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(human => human.sex === 'f');
+
+  const mothers = women.filter(woman =>
+    people.some(person => person.mother
+      === woman.name));
+
+  const womenWithChildren = withChildren
+    ? mothers
+    : women;
+
+  const average = womenWithChildren.reduce((acc, woman) => {
+    return acc + woman.died - woman.born;
+  }, 0) / womenWithChildren.length;
+
+  return average;
```

Comment:
you have the same logic in men function, please create a function helper for calculating average age